src/main/java/picocli/CommandLine.java

```diff
@@ -11625,10 +11649,15 @@ public String getString(String key, String defaultValue) {
             public String[] getStringArray(String key, String[] defaultValues) {
                 if (isEmpty()) { return defaultValues; }
                 String cmd = spec.qualifiedName(".");
-                List<String> result = addAllWithPrefix(rb, cmd + "." + key, keys, new ArrayList<String>());
-                if (!result.isEmpty()) { return result.toArray(new String[0]); }
-                addAllWithPrefix(rb, key, keys, result);
-                return result.isEmpty() ? defaultValues : result.toArray(new String[0]);
+                String qualifiedKey = cmd + "." + key;
+                String[] result = getStringArrayForExactKey(qualifiedKey);
+                if (result==null) { result = getStringArrayForExactKey(key); }
+                return result!=null ? result : defaultValues;
+            }
+            private String[] getStringArrayForExactKey(String key) {
+            	List<String> result = addAllWithPrefix(rb, key, keys, new ArrayList<String>());
+            	if (!result.isEmpty()) { return result.toArray(new String[0]); }
+            	return parent()==null ? null : parent().getStringArrayForExactKey(key);
```

Comment:
please add spaces around `==`

Reply:
> I found more whitespace style issues. Sorry to be nitpicking, but I would prefer to keep the code style consistent. 😓 Can you take a look?

No worries at all. You give clear instructions and I definitely overlooked those issues in the file. I've got no issues with getting a bit of tough love. 😉👍

Reply:
Done.

Reply:
Done.